allel/io/vcf_read.py

```diff
@@ -421,6 +433,10 @@ def vcf_to_npz(input, output,
         transformers=transformers
     )
 
+    if data is None:
+        # no data, bail out
```

Comment:
Should we provide a warning here?

Reply:
I also went back-and-forth on that :-) In the end I figured there might be cases where you're iterating through a load of regions, and you expect in many cases there may be no variants for some reasons, in which case you would get a lot of warnings and that would be annoying.

Reply:
I also felt that for novice users it might be helpful to get a descriptive warning if there a no variants. So hard to decide how to balance this. However I figured returning None if no variants should be a reasonably good clue as to there being no variants.

Reply:
Seems very reasonable! It's clearly stated in the docstring too